modyn/storage/internal/file_wrapper/csv_file_wrapper.py

```diff
@@ -0,0 +1,198 @@
+import csv
+from typing import Iterator, Optional
+
+from modyn.storage.internal.file_wrapper.abstract_file_wrapper import AbstractFileWrapper
+from modyn.storage.internal.file_wrapper.file_wrapper_type import FileWrapperType
+from modyn.storage.internal.filesystem_wrapper.abstract_filesystem_wrapper import AbstractFileSystemWrapper
+
+
+class CsvFileWrapper(AbstractFileWrapper):
+    def __init__(self, file_path: str, file_wrapper_config: dict, filesystem_wrapper: AbstractFileSystemWrapper):
+        super().__init__(file_path, file_wrapper_config, filesystem_wrapper)
+
+        self.file_wrapper_type = FileWrapperType.CsvFileWrapper
+
+        if "separator" in file_wrapper_config:
+            self.separator = file_wrapper_config["separator"]
+        else:
+            self.separator = ";"
+
+        if "label_index" not in file_wrapper_config:
+            raise ValueError(
+                "Please specify the index of the column that contains the label. "
+                "Use None if no column contains the label"
```

Comment:
Right now we assume we always have labels so this should be schema-enforced

Reply:
OK, but it is only a CSV file wrapper parameter. We have already discussed something similar in STB, we should either use conditional dependencies in the schema or raise the ValueError as here

Reply:
Ahh true, sorry, yeah then let's enforce it in init (that it is there and not None)

Reply:
Yep, now it checks that the value is available and that is a positive integer